dev/release.sh

```diff
@@ -0,0 +1,614 @@
+            if [ -z "$next_method2" ]; then
+                next_method2=$next_method
+            fi
+            shift
+            if [ "$next_method" = 'final' ]; then
+                do_branch=true
+            fi
+            ;;
+        --next-beta )
+            next_method2=$(echo "x$1" | sed -e 's|^x--next-||')
+            shift
+            ;;
+        --branch )
+            do_branch=true
+            shift
+            ;;
+        --no-upload )
+            do_upload=false
+            shift
+            ;;
+        --no-update )
+            do_update=false
+            shift
+            ;;
+        --verbose )
+            VERBOSE=echo
+            git_quiet=
+            shift
+            ;;
+        --debug )
+            DEBUG=echo
+            do_upload=false
+            shift
+            ;;
+        --force )
+            force=true
+            shift
+            ;;
+        --help )
+            usage
+            exit 0
+            ;;
+        --manual )
+            sed -e '1,/^### BEGIN MANUAL/d' \
+                -e '/^### END MANUAL/,$d' \
+                < "$0" \
+                | pod2man \
+                | man -l -
+            exit 0
+            ;;
+        -- )
+            shift
+            break
+            ;;
+        * )
+            echo >&2 "Unknown option $1"
+            shift
+            exit 1
+            ;;
+    esac
+done
+
+$DEBUG >&2 "DEBUG: \$next_method=$next_method"
+$DEBUG >&2 "DEBUG: \$next_method2=$next_method2"
+
+$DEBUG >&2 "DEBUG: \$do_branch=$do_branch"
+
+$DEBUG >&2 "DEBUG: \$do_upload=$do_upload"
+$DEBUG >&2 "DEBUG: \$do_update=$do_update"
+$DEBUG >&2 "DEBUG: \$DEBUG=$DEBUG"
+$DEBUG >&2 "DEBUG: \$VERBOSE=$VERBOSE"
+$DEBUG >&2 "DEBUG: \$git_quiet=$git_quiet"
+
+case "$next_method+$next_method2" in
+    major+major | minor+minor )
+        # These are expected
+        ;;
+    alpha+alpha | alpha+beta | beta+beta | final+final | + | +beta )
+        # These are expected
+        ;;
+    * )
+        echo >&2 "Internal option error ($next_method, $next_method2)"
+        exit 1
+        ;;
+esac
+
+# Setup ##############################################################
+
+# Make sure we're in the work directory
+cd $(dirname $0)/..
+HERE=$(pwd)
+
+# Check that we have the scripts that define functions we use
+found=true
+for fn in "./dev/release-aux/release-version-fn.sh" \
+          "./dev/release-aux/release-state-fn.sh"; do
+    if ! [ -f "$fn" ]; then
+        echo >&2 "'$fn' is missing"
+        found=false
+    fi
+done
+if ! $found; then
+    exit 1
+fi
+
+# Load version functions
+. ./dev/release-aux/release-version-fn.sh
+. ./dev/release-aux/release-state-fn.sh
+
+# Make sure it's a branch we recognise
+current_branch=$(git rev-parse --abbrev-ref HEAD)
+if (echo "current_branch" \
+        | grep -E -q \
+               -e '^master$' \
+               -e '^OpenSSL_[0-9]+_[0-9]+_[0-9]+[a-z]*-stable$' \
+               -e '^openssl_[0-9]+\.[0-9]+\.x$'); then
```

Comment:
Why .x?

Reply:
It's a branch that's going to have all the PATCH releases, just like we have letter releases in pre-3.0.  `x` represents the series of PATCH numbers, and distinguishes the *branch* from releases that would contain numbers only.

I was inspired by other projects I've been involved in which do exactly this.  https://github.com/darktable-org/darktable is one example.